addons/mail/static/src/models/messaging_notification_handler.js

```diff
@@ -446,12 +460,21 @@ registerModel({
          * @param {integer[]} param0.messag_ids
          */
         _handleNotificationMessageDelete({ message_ids }) {
+            let decrementAmounts = new Array(this.mailboxInfos.length).fill(0);
```

Comment:
Note that generally speaking it's not a good idea to update counters based on flux specific information like this one. If the deletion of the message is known through another means it will make the counter unsync and obsolete.

Also if the counter somehow did not yet take into account that message (everything is very async and there are race conditions possible) then it would decrement it incorrectly.

But it is acceptable for now.

Reply:
Yes indeed, I understand that. Especially if there are a mix of absolute and relative update. Then the order matter a lot. Actually, I started by doing it with absolute update but I changed my mind because it might be a lot of queries for the server if a lot of document are deleted.
(I have kept the implementation if needed)